frontend/src/data/variableProviders.ts

```diff
@@ -67,16 +68,21 @@ export function getProvider(variableId: MetricId): VariableProvider {
  * Returns a list of all VariableProviders required to get the specified
  * variables.
  */
-export function getUniqueProviders(
-  variableIds: MetricId[]
-): VariableProvider[] {
-  const providerIds = variableIds.map((id) => metricsToProviderIds[id]);
+export function getUniqueProviders(metricIds: MetricId[]): VariableProvider[] {
+  const providerIds = metricIds.map((id) => metricsToProviderIds[id]);
   const dedupedIds = Array.from(new Set(providerIds));
   return dedupedIds.map((id) => providersById[id]);
 }
 
-/** Returns the list of dataset ids that the provided variables depend on. */
-export function getDependentDatasets(variableIds: MetricId[]): string[] {
-  const providers = variableIds.map((id) => getProvider(id));
-  return VariableProvider.getUniqueDatasetIds(providers);
+/** Returns the list of dataset ids that the provided MetricQuerys depend on. */
+export function getDependentDatasets(queries: MetricQuery[]): string[] {
+  let datasetIds = new Set<string>();
+  queries.forEach((query) =>
+    query.metricIds.forEach((id) =>
+      getProvider(id)
+        .getRequiredDatasetIds(query)
+        .forEach((id) => datasetIds.add(id))
+    )
+  );
+  return Array.from(datasetIds);
```

Comment:
nit: I prefer a slightly more functional style for this kind of logic since it's more concise, similar to the way `getUniqueDatasetIds` works. But up to you, I think this is more of a style preference

Reply:
dicsussed on slack